src/main/java/net/dv8tion/jda/core/requests/restaction/InviteAction.java

```diff
@@ -0,0 +1,107 @@
+package net.dv8tion.jda.core.requests.restaction;
+
+import net.dv8tion.jda.core.JDA;
+import net.dv8tion.jda.core.entities.EntityBuilder;
+import net.dv8tion.jda.core.entities.Invite;
+import net.dv8tion.jda.core.requests.*;
+
+import org.json.JSONObject;
+
+public class InviteAction extends RestAction<Invite>
+{
+    private Integer maxAge = null;
+    private Integer maxUses = null;
+    private Boolean temporary = null;
+    private Boolean unique = null;
+
+    public InviteAction(final JDA api, final String channelId)
+    {
+        super(api, Route.Invites.CREATE_INVITE.compile(channelId), null);
+    }
+
+    @Override
+    protected void finalizeData()
+    {
+        final JSONObject object = new JSONObject();
+
+        if (this.maxAge != null)
+            object.put("max_age", (int) this.maxAge);
+        if (this.maxUses != null)
+            object.put("max_uses", (int) this.maxUses);
+        if (this.temporary != null)
+            object.put("temporary", (boolean) this.temporary);
+        if (this.unique != null)
+            object.put("unique", (boolean) this.unique);
+
+        this.data = object;
+    }
+
+    @Override
+    protected void handleResponse(final Response response, final Request request)
+    {
+        if (response.isOk())
+            request.onSuccess(EntityBuilder.get(this.api).createInvite(response.getObject()));
+        else
+            request.onFailure(response);
+    }
+
+    /**
+     * Sets the max age in seconds for the invite. Set this to {@code 0} if the invite should never expire. Default is {@code 86400} (24 hours).
+     *
+     * @param  maxAge
+     *         The max age for this invite or {@code null} to use the default value.
+     *
+     * @return The current InviteAction for chaining.
+     */
+    public final InviteAction setMaxAge(final Integer maxAge)
+    {
+        if (maxAge < 0)
```

Comment:
Is there a maximum that can be reached?
The client limits to a few options here like `30min, 1hr, 6hrs, etc` does this work with any number?